### PR TITLE
Example apps

### PR DESCRIPTION
**Pull request by LinuxBozo**
A series of small bare minimum apps for showcasing each buildpack Docker image

### CODE REVIEW

**Comment by toolness** [comment]
Would it be useful to have some kind of integration test suite that spins all these apps up in their appropriate docker containers and ensures that they work OK?

**Comment by toolness** [comment]
Progress so far:

* The php app works, but "gracefully stopping" it times out, which is a little bit annoying. Solutions might be to somehow fix it on the buildpack side, or tell users to use the [`stop_signal` directive](https://docs.docker.com/compose/compose-file/#stopsignal) in their `docker-compose.yml`.

* The go app works just fine.

* The python app actually raises an exception on `docker-compose up`:

```
Creating python_app_1
Attaching to python_app_1
app_1  | [2017-04-25 21:28:08 +0000] [1] [INFO] Starting gunicorn 19.7.1
app_1  | Traceback (most recent call last):
app_1  |   File "/home/vcap/app/.cloudfoundry/python/bin/gunicorn", line 11, in
<module>
app_1  |     sys.exit(run())
app_1  |   File "/app/.cloudfoundry/python/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 74, in run
app_1  |     WSGIApplication("%(prog)s [OPTIONS] [APP_MODULE]").run()
app_1  |   File "/app/.cloudfoundry/python/lib/python2.7/site-packages/gunicorn/app/base.py", line 203, in run
app_1  |     super(Application, self).run()
app_1  |   File "/app/.cloudfoundry/python/lib/python2.7/site-packages/gunicorn/app/base.py", line 72, in run
app_1  |     Arbiter(self).run()
app_1  |   File "/app/.cloudfoundry/python/lib/python2.7/site-packages/gunicorn/arbiter.py", line 198, in run
app_1  |     self.start()
app_1  |   File "/app/.cloudfoundry/python/lib/python2.7/site-packages/gunicorn/arbiter.py", line 139, in start
app_1  |     self.pidfile.create(self.pid)
app_1  |   File "/app/.cloudfoundry/python/lib/python2.7/site-packages/gunicorn/pidfile.py", line 39, in create
app_1  |     os.rename(fname, self.fname)
app_1  | OSError: [Errno 26] Text file busy
python_app_1 exited with code 1
```

That's weird.  I actually ran into something similar when [dockerizing `usaspending-api`](https://github.com/fedspendingtransparency/usaspending-api/pull/306) earlier today so this might be something wrong on my system...

File: examples/ruby/Gemfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+source 'http://rubygems.org'
```
Comment (afeld) [inline examples/ruby/Gemfile]: `https` 😉

**Review by afeld — COMMENTED** [review]
Just looked at the Ruby example, as requested - looks good!

**Comment by toolness** [comment]
Ok, I just added a `manage_examples.py` script which lets you test them.

Try e.g. `manage_examples.py test go` to test just the go example, or `manage_examples.py test` to test everything.

**Comment by afeld** [comment]
I'd suggest we merge this sooner than later, and smaller changes/fixes can be done after.

**Comment by LinuxBozo** [comment]
@afeld Just needing an approved review. This is pretty much ready to go now.

**Review by toolness — APPROVED** [review]
Nice, this looks good to me, thanks!

**Pull request merged**